pyccel/ast/datatypes.py

```diff
@@ -989,7 +989,56 @@ def datatype(self):
 
         The datatype of the object.
         """
-        return self._index_type.datatype
+        return self._key_type.datatype
+
+    @property
+    def key_type(self):
+        """
+        The type of the keys of the object.
+
+        The type of the keys of the object.
+        """
+        return self._key_type
+
+    @property
+    def value_type(self):
+        """
+        The type of the values of the object.
+
+        The type of the values of the object.
+        """
+        return self._value_type
+
+    @property
+    def container_rank(self):
+        """
+        Number of dimensions of the container.
+
+        Number of dimensions of the object described by the container. This is
+        equal to the number of values required to index an element of this container.
+        """
+        return 1
```

Comment:
same question

Reply:
What would you expect this to be?

Reply:
answered in the previous question

Reply:
I disagree. This question is unrelated to the previous question as this one is about `container_rank` while the other is about the more general `rank`